src/api/search/src/bin/validation.js

```diff
@@ -31,8 +31,73 @@ const queryValidationRules = [
     .bail(),
 ];
 
-const validateQuery = (rules) => {
+/**
+ * Advanced search is more flexible, only needs at least ONE field, but can run without any too.
+ * Date formats must be YYYY-MM-DD
+ */
+const advancedQueryValidationRules = [
+  oneOf([
+    check('postText')
+      .exists({ checkFalsy: true })
+      .withMessage('post should not be empty')
+      .bail()
+      .isLength({ max: 256, min: 1 })
+      .withMessage('post should be between 1 to 256 characters')
+      .bail(),
+    check('authorText')
+      .exists({ checkFalsy: true })
+      .withMessage('author should exist')
+      .bail()
+      .isLength({ max: 100, min: 2 })
+      .withMessage('invalid author value')
+      .bail(),
+
+    check('dateEnd')
+      .exists({ checkFalsy: true })
+      .withMessage('dateEnd should exist')
+      .bail()
+      .isLength({ max: 100, min: 2 })
+      .withMessage('invalid author value')
+      .bail()
+      .isISO8601()
+      .withMessage('invalid date format')
+      .bail(),
+
+    check('dateStart')
+      .exists({ checkFalsy: true })
+      .withMessage('dateEnd should exist')
+      .bail()
+      .isLength({ max: 100, min: 2 })
+      .withMessage('invalid author value')
+      .bail()
+      .isISO8601()
+      .withMessage('invalid date format')
+      .bail(),
+
+    check('perPage')
```

Comment:
The values above seem separate from these next two, where you are providing a modifier of the results vs. a filter used to produce those results.  I would break them out of your `oneOf` logic.

Reply:
I fixed this, but I want to know is the search going to be useful for just date range? should we leverage that... my test right now I fixed so that you need some context (author,title or post) AND date. Let me know we can play around with this.

Reply:
I think requiring at least one of `author`, `title`, `post`, or the dates is useful.  The `perPage` stuff is separate, since it governs the results of any of those.